tests/e2e/vmgroups_test.go

```diff
@@ -431,3 +432,144 @@ func (vg *VmGroupTest) TestVmGroupVolumeClone(c *C) {
 
 	misc.LogTestEnd(c.TestName())
 }
+
+// test to verify volume is removed after user created vmgroup is removed with "--remove-volume" option
+// 1. create a user created vmgroup "vmgroup_test1" and add VMs to this vmgroup
+// 2. create two volumes for this vmgroup, one is on default datastore, and the other is on non default_datastore
+// 3. verify mount/unmount for those two volumes
+// 4. after unmount, verify volume is detached for those two volumes
+// 5. remove VMs from this vmgroup, and remove the vmgroup with "--remove-volume" option
+// 6. verify that volumes do not exist after removing vmgroup
+func (vg *VmGroupTest) TestVmgroupRemoveWithRemoveVol(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create a volume in vgTestVMgroup1 on default_datastore
+	vg.createVolumes(c, vg.volName1)
+
+	// Add create privilege for the second datastore
+	out, err := adminutils.AddCreateAccessForVMgroup(vg.config.EsxHost, vgTestVMgroup1, vg.config.Datastores[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Create another volume on second datastore
+	// need to use full name for volume name
+	vg.createVolumes(c, vg.volName2+"@"+vg.config.Datastores[1])
+
+	// Verify volume can be mounted and unmounted for the first volume
```

Comment:
As per the inline comment, please add verification step when volume was attached.